src/main/java/us/ihmc/commons/thread/RepeatingTaskThread.java

```diff
@@ -0,0 +1,390 @@
+    * The loop will be exited, and the thread will die.
+    * Same as calling {@link #kill()} then {@link #join()}.
+    * InterruptedExceptions are ignored. To handle interrupted exceptions,
+    * call {@link #kill()} then {@link #join()} manually.
+    */
+   public void blockingKill()
+   {
+      kill();
+      try
+      {
+         join();
+      }
+      catch (InterruptedException ignored) {}
+   }
+
+   /**
+    * The method that is executed repeatedly in a loop.
+    * <p>
+    * You may {@code @Override} this method with the code to repeat.
+    * Alternatively, if a {@link RunnableThatThrows} was passed it, this method
+    * will call the run method in the loop.
+    *
+    * @throws Throwable Any throwable that the overriding code or the passed in task may throw.
+    *       This throwable will be handled by the passed in {@link ExceptionHandler}
+    *       (by default it is {@link DefaultExceptionHandler#MESSAGE_AND_STACKTRACE}).
+    */
+   protected void repeat() throws Throwable
+   {
+      if (task != null)
+         task.run();
+   }
+
+   /**
+    * The {@link Thread#run()} method, overridden to run a loop.
+    * To extend this class {@link Thread} style, override {@link #repeat()} instead.
+    * <p>
+    * DO NOT CALL THIS METHOD. Well, you can, but why would you?
+    * You are using a thread to run things asynchronously, but calling this would run the loop synchronously.
+    * Why would you want that?
+    * <p>
+    * This method is a necessary evil committed for this class to extend Thread.
+    */
+   @Override
+   public final void run()
+   {
+      while (running)
+      {
+         try
+         {
+            synchronized (loopLock)
+            {  // No more runs remaining -> wait until something changes
+               if (remainingRepetitions == 0)
+               {
+                  loopLock.wait();
+                  continue;
+               }
+
+               // Decrement the counter for the run that's about to occur
+               if (remainingRepetitions > 0)
+                  remainingRepetitions--;
+            }
+
+            // If a period/frequency limit was set, wait until loop can run.
+            if (loopPeriodLowerLimit > 0.0)
+            {
+               /*
+                * This call must not swallow interrupts.
+                * As of writing this comment (Nov, 2024), LockSupport.parkNanos() is used internally to block.
+                * Although the throttler will block until the period has elapsed, the thread
+                * remains interrupted.
+                */
+               throttler.waitAndRun(loopPeriodLowerLimit);
+            }
+         }
+         catch (InterruptedException interrupted)
+         {  // Maintain interrupted status so that runInLoop can handle it
+            interrupt();
+         }
+
+         // Run the runInLoop method, and handle any exception it may throw.
+         ExceptionTools.handle(this::repeat, exceptionHandler);
```

Comment:
This should just be `task.run()` where the exception handling is built in the constructor. Then you can delete the repeat/runTask method.

Reply:
The `runTask` method is needed so it can be overridden

Reply:
Okay that sounds good